### DISCUSSION TITLE
Problem generalising 01. PyTorch Workflow Fundamentals model_0 to different data

### DISCUSSION THREAD

**Discussion by amadanmath** [mrdbourke/pytorch-deep-learning#374] [General]
I have tried to play with the `model_0` from https://www.learnpytorch.io/01_pytorch_workflow/
When I execute it as written (except for increasing `epochs`), I can see both training and evaluation shrink, and the parameters end up very close to the gold around 170th epoch:

    Epoch: 170 | MAE Train Loss: 0.008932482451200485 | MAE Test Loss: 0.005023092031478882
    OrderedDict([('weights', tensor([0.6951])), ('bias', tensor([0.2993]))])

For the record, here are the dataset and gold parameters used:

    # Create *known* parameters
    weight = 0.7
    bias = 0.3
    
    # Create data
    start = 0
    end = 1
    step = 0.02

However, when I just change the dataset parameters to this:

    start = 1
    end = 101
    step = 1

the loss and the state dictionary stabilise to this, around 360th epoch:

    Epoch: 360 | MAE Train Loss: 12.168511390686035 | MAE Test Loss: 9.61581039428711
    OrderedDict([('weights', tensor([0.8074])), ('bias', tensor([0.1928]))])

which is not really close to the gold parameter values. Why was the training unsuccessful here, and what else should be adjusted in order for this to work as expected? Does the training process assume the input data is between 0 and 1? If so, which part of the training loop code is sensitive to this? (It may be somewhere later in the lessons, but I have not yet gotten there.)

It seems to only happen when `X` is large; when I made `bias` and `weight` larger, so that `y` was large, the training converged to the correct answer (though I had to increase `epochs` and learning rate in order to reach it). The thing I don't understand is why making `X` larger made the result to stabilise on a wrong value (as opposed to e.g. not converge at all).

**Comment by PastorJordi** [comment]
I had exactly the same problem when playing around with the exercises.
Try standardizing your data to see whether it helps :)

[Here's the SO solution](https://stackoverflow.com/a/74930159/16173298) I found that helped me
